clusterMembership/dynamicCluster.go

```diff
@@ -0,0 +1,72 @@
+package clusterMembership
+
+import "fmt"
+import "errors"
+
+/*
+ * Represents a Cluster of nodes that can change over time.
+ * Keeps a Map of the nodeId to Node for fast lookup of nodes.
+ */
+type dynamicCluster struct {
+	members map[string]Node
+}
+
+/*
+ * Returns a Snapshot of the Cluster Membership state
+ */
+func (c dynamicCluster) GetMembers() []string {
+
+	var nodes = make([]string, len(c.members))
+	var index = 0
+
+	for _, n := range c.members {
+		nodes[index] = n.GetId()
+		index++
+	}
+
+	return nodes
+}
+
+/*
+ * Adds A Node from the Cluster
+ */
+func (c dynamicCluster) AddNode(n Node) {
```

Comment:
Yes, so my plan is to run ClusterMembership in its own goroutine.  This way all updates to the underlying cluster data structure are single threaded.